### PR TITLE
COP-5459: Create schema for groups table

### PR DESCRIPTION
**Pull request by rfndn**
Create schema for the new groups functionality outlined in COP-5459

### CODE REVIEW

**Review by pixie79 — CHANGES_REQUESTED** [review]
This should not exist

File: flyway/schemas/reference/V137__groups.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+-- Table comment
+COMMENT ON TABLE groups IS '{"label": "Groups", "description": "A list of Border Force Groups", "owner": "cop@homeoffice.gov.uk", "schemalastupdated": "20/04/2021", "dataversion": 1}';
+-- Column comments
+COMMENT ON COLUMN groups.id IS '{"label": "Identifier", "description": "Database unique identity record.", "summaryview": "false"}';
+COMMENT ON COLUMN groups.name IS '{"label": "Name", "description": "The name of the group.", "businesskey": true, "summaryview": "true"}';
+COMMENT ON COLUMN groups.keycloakpath IS '{"label": "Keycloak path", "description": "The name of the group.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.displayname IS '{"label": "Display name", "description": "User friendly name of the group.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.grouptype IS '{"label": "Group Type", "description": "The type of group which is either Team, Role, Location.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.code IS '{"label": "Code", "description": "The code of the group. This corresponds to the team code if Group Type is Team.", "summaryview": "true"}';
```
Comment (pixie79) [inline flyway/schemas/reference/V137__groups.sql:25]: Code makes no sense if you have name as the businessKey

File: flyway/schemas/reference/V137__groups.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+COMMENT ON COLUMN groups.teamid IS '{"label": "Team ID", "description": "UUID of the Team which the Group relates to if applicable.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.branchid IS '{"label": "Branch", "description": "Branch of the group.", "summaryview": "true"}';
```
Comment (pixie79) [inline flyway/schemas/reference/V137__groups.sql:27]: One group can be part of many branches

File: flyway/schemas/reference/V137__groups.sql [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+COMMENT ON COLUMN groups.locationid IS '{"label": "Location", "description": "Location of the group.", "summaryview": "true"}';
```
Comment (pixie79) [inline flyway/schemas/reference/V137__groups.sql:28]: One group can be part of many locations

File: flyway/schemas/reference/V137__groups.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  validfrom TIMESTAMP WITH TIME ZONE,
+  validto TIMESTAMP WITH TIME ZONE,
+  updatedby VARCHAR(60) NULL
+);
+
+-- Table comment
+COMMENT ON TABLE groups IS '{"label": "Groups", "description": "A list of Border Force Groups", "owner": "cop@homeoffice.gov.uk", "schemalastupdated": "20/04/2021", "dataversion": 1}';
+-- Column comments
+COMMENT ON COLUMN groups.id IS '{"label": "Identifier", "description": "Database unique identity record.", "summaryview": "false"}';
+COMMENT ON COLUMN groups.name IS '{"label": "Name", "description": "The name of the group.", "businesskey": true, "summaryview": "true"}';
```
Comment (pixie79) [inline flyway/schemas/reference/V137__groups.sql:21]: Should this not be group code - there is no point in name / code and display name. Code and display name is better

File: flyway/schemas/reference/V137__groups.sql [hunk 5]
```diff
@@ -0,0 +1,36 @@
+CREATE TABLE groups (
+  id UUID NOT NULL PRIMARY KEY,
+  name VARCHAR(60) NOT NULL,
+  keycloakpath VARCHAR(60) NOT NULL,
+  displayname VARCHAR(60) NOT NULL,
```
Comment (pixie79) [inline flyway/schemas/reference/V137__groups.sql:5]: How long is too long for the screen to make sense - 60 for a group names seems a bit big. Does the UI handle wrapping?

File: flyway/schemas/reference/V137__groups.sql [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+-- Column comments
+COMMENT ON COLUMN groups.id IS '{"label": "Identifier", "description": "Database unique identity record.", "summaryview": "false"}';
+COMMENT ON COLUMN groups.name IS '{"label": "Name", "description": "The name of the group.", "businesskey": true, "summaryview": "true"}';
+COMMENT ON COLUMN groups.keycloakpath IS '{"label": "Keycloak path", "description": "The name of the group.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.displayname IS '{"label": "Display name", "description": "User friendly name of the group.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.grouptype IS '{"label": "Group Type", "description": "The type of group which is either Team, Role, Location.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.code IS '{"label": "Code", "description": "The code of the group. This corresponds to the team code if Group Type is Team.", "summaryview": "true"}';
+COMMENT ON COLUMN groups.teamid IS '{"label": "Team ID", "description": "UUID of the Team which the Group relates to if applicable.", "summaryview": "true"}';
```
[the remark this answers is not in the document]
Comment (aminmc) [reply flyway/schemas/reference/V137__groups.sql:26]: why is this called teamid? Which is it...is it a group or is it a team?

File: flyway/schemas/reference/V137__groups.sql (hunk 5 above)
Comment (rfndn) [reply flyway/schemas/reference/V137__groups.sql:5]: It's up to each individual app consuming this data app to handle the wrapping, this repo shouldn't have anything to do with the way the data is displayed.

File: flyway/schemas/reference/V137__groups.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (rfndn) [reply flyway/schemas/reference/V137__groups.sql]: That's not a business requirement.

File: flyway/schemas/reference/V137__groups.sql (hunk 4 above)
Comment (rfndn) [reply flyway/schemas/reference/V137__groups.sql]: There was a clear business requirement for a separate code, group name and group display name.

**Comment by dolomitejunkie** [comment]
looks good to me with the latest changes

**Comment by dolomitejunkie** [comment]
all looks good ot me, many thanks

**Pull request closed without merging**